### PR TITLE
lib/preview: functionalize

### PR DESCRIPTION
**Pull request by gaelicWizard**
## Description
Refactor in to a function, but continue to support current usage.

## Motivation and Context
This allows running the preview without reloading the startup files multiple times, and alsö integrate to `bash-it` function, and alsö alsö offer it's own completions. 

## How Has This Been Tested?
Tested locally and all tests pass.

## Types of changes
- [x] New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code follows the code style of this project.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] If I have added a new file, I also added it to ``clean_files.txt`` and formatted it using ``lint_clean_files.sh``.
- [ ] I have added tests to cover my changes, and all the new and existing tests pass.

### CODE REVIEW

File: lib/preview.bash [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
+#
+# Displays the prompt from each _Bash It_ theme.
 
-  Previewing Bash-it Themes
+function _bash-it-preview() {
+	printf '\n\n\t%s\n\n' "Previewing Bash-it Themes"
 
-  "
+	local themes theme
+	themes=("${1:-"${BASH_IT}/themes"/*/*.theme.bash}")
```
Comment (NoahGorny) [inline lib/preview.bash:9]: shouldnt you use `themes"/**/*.theme.bash`

File: lib/preview.bash [hunk 2]
```diff
@@ -1,19 +1,20 @@
-if [[ "${BASH_PREVIEW:-}" ]];
-then
-  unset BASH_PREVIEW #Prevent infinite looping
-  echo "
+# shellcheck shell=bash
+#
+# Displays the prompt from each _Bash It_ theme.
 
-  Previewing Bash-it Themes
+function _bash-it-preview() {
```
Comment (NoahGorny) [inline lib/preview.bash:5]: maybe we should provide a way to do this from the command line? or at least doc about it?

**Review by NoahGorny — CHANGES_REQUESTED** [review]

File: lib/preview.bash (hunk 2 above)
Comment (gaelicWizard) [reply lib/preview.bash:5]: Absolutely! In fact, that's the motivation for the change. But it's extremely hard to make non-conflicting changes to `bash-it`, so I left that for a subsequent PR after all these other changes are eventually merged (including your `profile` subcommand).

Comment (NoahGorny) [reply lib/preview.bash:5]: lets try to have this as part of this PR!

Comment (gaelicWizard) [reply lib/preview.bash:5]: Ok, I've added it to this PR and even updated the documentation.

File: lib/helpers.bash [hunk 3]
```diff
@@ -95,7 +95,7 @@ alias reload_plugins="$(_make_reload_alias plugin plugins)"
 bash-it ()
 {
     about 'Bash-it help and maintenance'
-    param '1: verb [one of: help | show | enable | disable | migrate | update | search | version | reload | restart | doctor ] '
+    param '1: verb [one of: help | show | enable | disable | migrate | update | search | preview | version | reload | restart | doctor ] '
```
Comment (NoahGorny) [inline lib/helpers.bash:98]: you should add this to bash-it completion as well!

Comment (gaelicWizard) [reply lib/helpers.bash:98]: ok

Comment (gaelicWizard) [reply lib/helpers.bash:98]: Done and doen

File: lib/helpers.bash [hunk 4]
```diff
@@ -105,6 +105,8 @@ bash-it ()
     example '$ bash-it migrate'
     example '$ bash-it update'
     example '$ bash-it search [-|@]term1 [-|@]term2 ... [ -e/--enable ] [ -d/--disable ] [ -r/--refresh ] [ -c/--no-color ]'
+	example '$ bash-it preview'
```
Comment (NoahGorny) [inline lib/helpers.bash:108]: sorry about the nit-pick , but it seems like the spacing here is not like in the rest of the file 😢

Comment (gaelicWizard) [reply lib/helpers.bash:108]: Until my `helpers` branch is committed, `lib/helpers` has not been through `shfmt` so the spacing varies between functions even...

**Comment by gaelicWizard** [comment]
I've expanded the `_bash-it-preview()` function to offer it's own completions, theme names!

**Review by NoahGorny — APPROVED** [review]
LGTM

**Comment by gaelicWizard** [comment]
Rebased on current master and ready to go!

**Pull request merged**